fe/fe-core/src/main/java/org/apache/doris/binlog/DBBinlog.java

```diff
@@ -340,7 +338,11 @@ private BinlogTombstone dbBinlogDisableGc() {
             if (tombstone != null) {
                 tombstones.add(tombstone);
             }
+            if (tableBinlog.isEmpty()) {
+                tableBinlogMap.remove(tableBinlog.getTableId());
+            }
```

Comment:
Acquire lock before checking `tableBinlog.isEmpty` and updating `tableBinlogMap`.